src/main/java/org/betonquest/betonquest/BetonQuest.java

```diff
@@ -343,6 +336,17 @@ public class BetonQuest extends JavaPlugin {
      */
     private LastExecutionCache lastExecutionCache;
 
+    /**
+     * The BetonQuest plugin constructor.
+     */
+    @SuppressWarnings("PMD.AssignmentToNonFinalStatic")
+    public BetonQuest() {
+        super();
+        instance = this;
+        log = BetonQuestLogger.create(this);
```

Comment:
It is not a good idea at all to ignore this warning. It would be much better to move the logger initialization to the beginning of `onEnable` instead of to the constructor.